### ISSUE TITLE
Customize/Override default scss theme?

### ISSUE THREAD

**Issue by arnlen** [miguelcobain/ember-paper#193]
I see in the sources that the project is designed to work with theme, but I don't see any way to see a `THEME` variable somewhere.

What is the proper way to override the default theme?
Is there a way to activate my own theme after I've set the `theme-name` in my style file like following:

```scss
//app/styles/custom.scss

$theme-name: "my-theme";

$primary: "red";
$accent: "yellow";
$warn: "brown";
$background: "red";
$link: "blue";
...
```

**Comment by miguelcobain** [comment]
`$theme-name` is not currently used for that purpose.

To create a custom theme, simply override the desired palletes.
You can even create new ones.

**Comment by arnlen** [comment]
Ok thanks for your answer.
Do you have plan to integrate sort of config file, in order to select the active theme?

**Comment by miguelcobain** [comment]
Do you have any proposal?
For me a theme is a collection of color palettes set to $primary, $accent, $warn, etc.

Any suggestion for improving this?

**Comment by zkwentz** [comment]
I have one, in `app/styles/color-palette.scss`

Could you merge `$color` map with a user defined one, if it's there?

Something like:

```scss
$colors: () !default;

...

$colors: map-merge($colors, (
  'red': $color-red,
  'pink': $color-pink,
  'purple': $color-purple,
  'deep-purple': $color-deep-purple,
  'indigo': $color-indigo,
  'blue': $color-blue,
  'light-blue': $color-light-blue,
  'cyan': $color-cyan,
  'teal': $color-teal,
  'green': $color-green,
  'light-green': $color-light-green,
  'lime': $color-lime,
  'yellow': $color-yellow,
  'amber': $color-amber,
  'orange': $color-orange,
  'deep-orange': $color-deep-orange,
  'brown': $color-brown,
  'blue-grey': $color-blue-grey,
  'grey': $color-grey,
  'shades': $shades,
  'foreground-light': $foreground-light,
  'foreground-dark': $foreground-dark
));
```

This would allow for us to make overrides, like the following (`app/styles/ember-paper-overrides.scss`):

```scss
$color-white: (
  '50': #fff,
  ...
);

$colors: (
  'white': $color-white,
  ...
);

$primary: 'white';
```

This could use some bikeshedding on the naming, but I think it's a good solution.

**Comment by DanChadwick** [comment]
@zkwentz -- I'm struggling with this today too. Another minor change would be to add `!default` to each `$color-xxxx` variable so that they can be overridden too, as in:
```scss
@import "color-palette";
$color-teal: map-merge($color-teal, ('contrast':  $light-contrast-color));
$primary: 'teal';
@import "ember-paper";
```

In this way, the second importing of the color palette withing `ember-paper.scss` wouldn't overwrite the modification.

Too bad the @import doesn't work like ES6 where it is imported only once.

**Comment by zkwentz** [comment]
@DanChadwick heh, yeah one `@import` we could do this with no change.

Great idea regarding `!default`ing all of the color maps.

**Comment by DanChadwick** [comment]
@zkwentz -- you want to make a PR or shall I? At least then we'll have something concrete to discuss. I think that the more `!defaults` in the config, the more flexible it will be.

Another idea would be to split out everything from `ember-paper.scss` except the `@imports`. In this way, if someone needs to copy it into their `app.scss`, they won't have to copy in anything except `@imports`.

**Comment by zkwentz** [comment]
Sure thing. I'll open one shortly.

**Comment by zkwentz** [comment]
@DanChadwick opened above ^.

Can you elaborate on this?

> Another idea would be to split out everything from ember-paper.scss except the @imports. In this way, if someone needs to copy it into their app.scss, they won't have to copy in anything except @imports.

**Comment by DanChadwick** [comment]
Currently `ember-paper.scss` has some font definitions and a few misc definitions, then at the end a series of `@import` directives. If the font and misc definitions were moved to a different file, leaving only the import directives, then someone could easily copy the contents of `ember-paper.scss` into their own `app.scss` file and insert customization as desired. If, say, the font or misc definitions change in the new file, they would not have to adjust their `app.scss`. Make sense?

**Comment by zkwentz** [comment]
Ah I see, yes that makes sense. :+1:

**Comment by DanChadwick** [comment]
This can be closed now as it's fixed in #301.
